cmd/create/cluster/cmd.go

```diff
@@ -493,37 +517,115 @@ func run(cmd *cobra.Command, _ []string) {
-			Help:     cmd.Flags().Lookup("compute-nodes").Usage,
-			Default:  computeNodes,
+	// Autoscaling
+	isAutoscalingSet := cmd.Flags().Changed("enable-autoscaling")
+	autoscaling := args.autoscalingEnabled
+	if !autoscaling && !isAutoscalingSet && interactive.Enabled() {
+		autoscaling, err = interactive.GetBool(interactive.Input{
+			Question: "Enable autoscaling",
+			Help:     cmd.Flags().Lookup("enable-autoscaling").Usage,
+			Default:  autoscaling,
+			Required: false,
 		})
 		if err != nil {
-			reporter.Errorf("Expected a valid number of compute nodes: %s", err)
+			reporter.Errorf("Expected a valid value for enable-autoscaling: %s", err)
 			os.Exit(1)
 		}
 	}
-	if multiAZ {
-		if computeNodes < 3 {
-			reporter.Errorf("The number of compute nodes needs to be at least 3")
+
+	isMinReplicasSet := cmd.Flags().Changed("min-replicas")
+	isMaxReplicasSet := cmd.Flags().Changed("max-replicas")
+
+	minReplicas := args.minReplicas
+	maxReplicas := args.maxReplicas
+	if autoscaling {
+
+		// if the user set compute-nodes and enabled autoscaling
+		if cmd.Flags().Changed("compute-nodes") {
+			reporter.Errorf("Compute-nodes can't be set when autoscaling is enabled")
 			os.Exit(1)
 		}
-		if computeNodes%3 != 0 {
+
+		if interactive.Enabled() || !isMinReplicasSet {
+			minReplicas, err = interactive.GetInt(interactive.Input{
+				Question: "Min replicas",
+				Help:     cmd.Flags().Lookup("min-replicas").Usage,
+				Default:  minReplicas,
+				Required: true,
+			})
+			if err != nil {
+				reporter.Errorf("Expected a valid number of min replicas: %s", err)
+				os.Exit(1)
+			}
+		}
+		if interactive.Enabled() || !isMaxReplicasSet {
+			maxReplicas, err = interactive.GetInt(interactive.Input{
+				Question: "Max replicas",
+				Help:     cmd.Flags().Lookup("max-replicas").Usage,
+				Default:  maxReplicas,
+				Required: true,
+			})
+			if err != nil {
+				reporter.Errorf("Expected a valid number of max replicas: %s", err)
+				os.Exit(1)
+			}
+		}
+
+		if minReplicas < 2 {
+			reporter.Errorf("Cluster requires at least 2 compute nodes")
+			os.Exit(1)
+		}
+
+		if minReplicas > maxReplicas {
+			reporter.Errorf("max-replicas must be greater or equal to min-replicas")
+			os.Exit(1)
+		}
+
+		if multiAZ && (minReplicas%3 != 0 || maxReplicas%3 != 0) {
 			reporter.Errorf("Multi AZ clusters require that the number of compute nodes be a multiple of 3")
 			os.Exit(1)
 		}
-	} else {
-		if computeNodes < 2 {
-			reporter.Errorf("The number of compute nodes needs to be at least 2")
+	}
+
+	// Compute nodes:
+	computeNodes := args.computeNodes
+	// Compute node requirements for multi-AZ clusters are higher
+	if multiAZ && !autoscaling && !cmd.Flags().Changed("compute-nodes") {
+		computeNodes = 3
+	}
+	if !autoscaling {
+		// if the user set min/max replicas and hasn't enabled autoscaling
+		if isMinReplicasSet || isMaxReplicasSet {
+			reporter.Errorf("Autoscaling must be enabled in order to set min and max replicas")
```

Comment:
In playing with this, if I do something like `rosa create cluster -c foo --min-replicas=2 --max-replicas=4` the user will probably expect the `--enable-autoscaling` to be set automatically, so maybe forcing interactive mode on the user to confirm that's what they want would be a good idea. What do you think?

Reply:
I can, though edit behavior would be different.

If the machinepool already has autoscaling enabled and the user only wants to set min replicas - they should be able set 
`rosa edit machinepool foo -c bar --min-replicas=3` without setting `--enable-autoscaling`. However if autoscaling is disabled I don't want to turn to interactive because I want to user to expilicitly enable it.

Reply:
On second thought, I don't want to go there.
If we start complicating validations - potential bugs are coming.
We will have to check also that the user didn't set `--min-replicas` with `--replicas`

If the user set `--enable-autoscaling` and not the min,max replicas they will get prompted for them, but I think we should keep the `enable-autoscaling flag explicit` (except for edit, when it is unchanged)

Reply:
this sounds reasonable